packages/strapi-plugin-i18n/controllers/content-types.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const { pick, uniq, prop, propOr, flatten, pipe } = require('lodash/fp');
+const { getService } = require('../utils');
+
+module.exports = {
+  async getNonLocalizedAttributes(ctx) {
+    const { user } = ctx.state;
+    const { model, id, locale } = ctx.request.body;
+
+    const modelDef = strapi.getModel(model);
+    const { copyNonLocalizedAttributes, isLocalized } = getService('content-types');
+    const { READ_ACTION, CREATE_ACTION } = strapi.admin.services.constants;
+
+    if (!isLocalized(modelDef)) {
+      return ctx.badRequest('model.not.localized');
+    }
+
+    const entity = await strapi.entityService.findOne({ params: { id } }, { model });
+
+    if (!entity) {
+      return ctx.notFound();
+    }
+
+    const permissions = await strapi.admin.services.permission.find({
+      action_in: [READ_ACTION, CREATE_ACTION],
+      subject: model,
+      role_in: user.roles.map(prop('id')),
+    });
+
+    const localePermissions = permissions
+      .filter(p => propOr([], 'properties.locales', p).includes(locale))
```

Comment:
you should create a getLocaleProperty so you can make the code more explicit
 (if not already available in @Convly PR)